lib/teiserver/player/tachyon_handler.ex

```diff
@@ -301,6 +321,172 @@ defmodule Teiserver.Player.TachyonHandler do
+         :ok <- Account.accept_friend_request(originator_id, state.user.id) do
+      Player.Session.friend_request_accepted(originator_id, state.user.id)
+      {:response, cmd_id, nil, state}
+    else
+      {:error, :invalid_id} ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :invalid_user)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+
+      {:error, "no request"} ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :no_pending_request)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+    end
+  end
+
+  def handle_command("friend/rejectRequest" = cmd_id, "request", message_id, msg, state) do
+    with {:ok, originator_id} <- parse_user_id(msg["data"]["from"]),
+         %Account.FriendRequest{} = req <-
+           Account.get_friend_request(originator_id, state.user.id),
+         {:ok, _changeset} <- Account.delete_friend_request(req) do
+      Player.Session.friend_request_rejected(originator_id, state.user.id)
+      {:response, cmd_id, nil, state}
+    else
+      {:error, :invalid_id} ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :invalid_user)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+
+      nil ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :no_pending_request)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+
+      err ->
+        Logger.error("Unhandled error in rejectRequest: #{inspect(err)}")
+
+        resp =
+          Schema.error_response(cmd_id, message_id, :internal_error, inspect(err))
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+    end
+  end
+
+  def handle_command("friend/cancelRequest" = cmd_id, "request", message_id, msg, state) do
+    with {:ok, target_id} <- parse_user_id(msg["data"]["to"]),
+         %Account.FriendRequest{} = req <- Account.get_friend_request(state.user.id, target_id),
+         {:ok, _changeset} <- Account.delete_friend_request(req) do
+      Player.Session.friend_request_cancelled(target_id, state.user.id)
+      {:response, cmd_id, nil, state}
+    else
+      _ ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :invalid_user)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
+    end
+  end
+
+  def handle_command("friend/remove" = cmd_id, "request", message_id, msg, state) do
+    with {:ok, target_id} <- parse_user_id(msg["data"]["userId"]),
+         %Account.Friend{} = friend <- Account.get_friend(state.user.id, target_id),
+         {:ok, _changeset} <- Account.delete_friend(friend) do
+      Player.Session.friend_removed(target_id, state.user.id)
+      {:response, cmd_id, nil, state}
+    else
+      nil ->
+        resp =
+          Schema.error_response(cmd_id, message_id, :not_in_friendlist)
+          |> Jason.encode!()
+
+        {:reply, :ok, {:text, resp}, state}
```

Comment:
I know the schema for this is already done but it does remind me of our discussion on party spec::
> party/leave should probably have not_in_party fail reason.
>> I don't think it makes sense. What would a client do with that information? most likely, it'll ignore it since the goal is achieved. Making it a no-op also makes it idempotent which is always a nice property to have.

I think the same applies here, we even have `Account.delete_friend/2` which returns :ok if the users are not friends.

Reply:
amended the commit to make it a no-op when there is no friend request.